src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1506,8 +1506,14 @@ public static function addImageToTemplate($objTemplate, $arrItem, $intMaxWidth=n
 
 		try
 		{
-			$src = \Image::create($arrItem['singleSRC'], $size)->executeResize()->getResizedPath();
-			$picture = \Picture::create($arrItem['singleSRC'], $size)->getTemplateData();
+			$src = \System::getContainer()->get('contao.image.image_factory')->create(TL_ROOT . '/' . $arrItem['singleSRC'], $size)->getUrl(TL_ROOT);
```

Comment:
Yes, IMO that’s the right way. The image factory doesn’t know about `TL_ROOT` and the resizer only knows where to store the image files.